src/ExtendedMessageEntensions.cs

```diff
@@ -42,23 +42,18 @@ public static void WriteFixed32BigEndianDelimitedTo(this IMessage message, Strea
         /// <returns>The parsed message.</returns>
         public static T ParseFixed32BigEndianDelimitedFrom<T>(this MessageParser<T> parser, Stream input) where T : IMessage<T>
         {
-            using (var ms = new MemoryStream())
+            var lengthBuffer = new byte[4];
+            input.Read(lengthBuffer, 0, lengthBuffer.Length);
```

Comment:
`Stream.Read` can return less than the requested length.  Since it so small 4 `ReadByte` might be okay.

Reply:
Current implementation assumes input stream is valid, how would you suggest to handle malicious stream?

Reply:
In has nothing to do with malicious stream.  If the stream is TCP then you could reading across two TCP packets.  The stream will return 2 bytes from the first packet, Then in the next call return 2 bytes from the second packet.

Read the docs on [Stream.Read](https://docs.microsoft.com/en-us/dotnet/api/system.io.stream.read?view=netframework-4.8)